csfieldguide/static/interactives/matrix-simplifier/js/matrix-simplifier.js

```diff
@@ -0,0 +1,493 @@
+  row0 = vsprintf(rowTemplate, matrix[0]);
+  row1 = vsprintf(rowTemplate, matrix[1]);
+  row2 = vsprintf(rowTemplate, matrix[2]);
+
+  return sprintf(MATRIX_TEMPLATE, row0, row1, row2);
+}
+
+
+/* Set the matrices in modal to the default values */
+function resetModalMatrices() {
+  // reset to default values of modal matrices
+  $('#matrix-row-0-col-0').val(1);
+  $('#matrix-row-0-col-1').val(0);
+  $('#matrix-row-0-col-2').val(0);
+
+  $('#matrix-row-1-col-0').val(0);
+  $('#matrix-row-1-col-1').val(1);
+  $('#matrix-row-1-col-2').val(0);
+
+  $('#matrix-row-2-col-0').val(0);
+  $('#matrix-row-2-col-1').val(0);
+  $('#matrix-row-2-col-2').val(1);
+
+  $('#vector-row-0').val(1);
+  $('#vector-row-1').val(0);
+  $('#vector-row-2').val(0);
+
+  // remove red borders on inputs that had errors and enable add button
+  $('.matrix-row input').removeClass('input-error');
+  $('.add-from-input').prop('disabled', false);
+}
+
+
+/**
+ * Calculates result of matrix multiplication and vector addition.
+ * Returns array containing result matrix and vector.
+ */
+function calculateOutput() {
+  var matrixResult = mathjs.zeros(3, 3);
+  var vectorResult = mathjs.zeros(3, 1);
+
+  if (currentMatricesOrder.length == 1) {
+    matrixResult = currentMatricesOrder[0];
+  } else if (currentMatricesOrder.length > 1) {
+    // 2 or more matrices
+    matricesCopy = currentMatricesOrder.slice(); // copy list for multiplying in place
+    matrixResult = multiplyMatrices(matricesCopy);
+  }
+
+  if (currentVectorsOrder.length == 1) {
+    vectorResult = currentVectorsOrder[0];
+  } else if (currentVectorsOrder.length > 1) {
+    // 2 or more vectors
+    vectorsCopy = currentVectorsOrder.slice(); // copy list for adding in place
+    vectorResult = addVectors(vectorsCopy);
+  }
+
+  return [matrixResult, vectorResult];
+}
+
+
+/**
+ * Multiply matrices together and return result
+ */
+function multiplyMatrices(m) {
+  var multiply = true;
+  while (multiply) {
+    if (m.length == 2) {
+      multiply = false;
+      return mathjs.multiply(m[0], m[1]);
+    } else {
+      result = mathjs.multiply(m[0], m[1]);
+      // remove the first matrix in array
+      m.shift();
+      // replace the new first element in array with result
+      m[0] = result;
+    }
+  }
+}
+
+
+/**
+ * Add vectors together and return result
+ */
+function addVectors(v) {
+  var add = true;
+  while (add) {
+    if (v.length == 2) {
+      add = false;
+      return mathjs.add(v[0], v[1]);
+    } else {
+      result = mathjs.add(v[0], v[1]);
+      // remove the first vector in array
+      v.shift();
+      // replace the new first element in array with result
+      v[0] = result;
+    }
+  }
+}
+
+
+/**
+ * Displays the output of calculations
+ */
+function showOutput() {
+  // If output container is showing, hide it while mathjax renders.
+  if (!$('#output-container').hasClass('invisible')) {
+    $('#output-container').addClass('invisible');
+  }
```

Comment:
It doesn't matter if the element already has the class, so you don't need the if statement